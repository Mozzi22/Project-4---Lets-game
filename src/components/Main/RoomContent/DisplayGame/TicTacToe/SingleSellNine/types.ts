export type TSingleSellNine = {
    id: string | number;
}