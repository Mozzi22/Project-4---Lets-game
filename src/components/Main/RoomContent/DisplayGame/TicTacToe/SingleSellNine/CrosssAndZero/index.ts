import CrossAndZero from './CrossAndZero';

export default CrossAndZero;
