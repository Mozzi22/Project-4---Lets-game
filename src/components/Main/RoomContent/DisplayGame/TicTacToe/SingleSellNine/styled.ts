import styled from 'styled-components';

export const StSingleSellNine = styled.div`
    width: 131px;
    height: 131px;
    background: white;
    cursor: pointer;
    color: black;
    border:  1px solid black;
    display: flex;
    justify-content: center;
    align-items: center;
`;
