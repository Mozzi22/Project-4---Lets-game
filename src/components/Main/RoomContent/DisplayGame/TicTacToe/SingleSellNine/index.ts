import SingleSellNine from './SingleSellNine';

export default SingleSellNine;
