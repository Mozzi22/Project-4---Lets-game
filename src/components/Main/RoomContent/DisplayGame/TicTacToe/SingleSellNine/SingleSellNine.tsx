import React from 'react';
import { StSingleSellNine } from './styled';
import { TSingleSellNine } from './types'

const SingleSellNine: React.FC<TSingleSellNine> = ({ id }) => {
    const handleClick = (e: React.MouseEvent) => {
        console.log(e.target.id);
    };

    return (
        <StSingleSellNine 
        id={id + 1} 
        onClick={handleClick}
        >
            {id + 1}
        </StSingleSellNine>
    );
};

export default SingleSellNine;
