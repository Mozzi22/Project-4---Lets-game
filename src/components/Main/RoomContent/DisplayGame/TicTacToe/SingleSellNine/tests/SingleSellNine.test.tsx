import React from 'react';
import SingleSellNine from '../SingleSellNine';
import { mount, shallow } from 'enzyme';
import configureStore from 'redux-mock-store';

const mockStore = configureStore();
const store = mockStore({
      ticTacToe: {}
});
describe('SingleSellNine', () => {
   const props = {
            id: 1,
            doStep: jest.fn(),
            status: '',
          };
    it('Should match snapshot', () => {
        const component = shallow(<SingleSellNine {...props}/>);
        expect(component.html()).toMatchSnapshot();
    });
    it('Should render SingleSellNine ', () => {
        const component = mount(<SingleSellNine {...props} />);
        expect(component.find('styled__StSingleSellNine')).toHaveLength(1);
    });
    it('Should render img CROSS ', () => {
        props.status = 'X';
        const component = mount(<SingleSellNine {...props} />);
        expect(component.find('styled__StSingleSellNine')).toBe('X');
    });
    it('Should render img ZEROTIC ', () => {
        props.status = 'O';
        const component = mount(<SingleSellNine {...props} />);
        expect(component.find('styled__StSingleSellNine')).toBe('O');
    });
    it('Should click tic-tac-toe field and call doStep ', () => {
        props.status = '';
        const component = mount(<SingleSellNine {...props} />);
        component.find('styled__StSingleSellNine').props().onClick();
        expect(props.doStep).toHaveBeenCalledWith(props.id);
    });
});
