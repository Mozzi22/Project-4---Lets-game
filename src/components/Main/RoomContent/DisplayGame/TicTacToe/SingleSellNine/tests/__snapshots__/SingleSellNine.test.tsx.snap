// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleSellNine Should match snapshot 1`] = `"<div id=\\"1\\" class=\\"styled__StSingleSellNine-sc-1icsnrc-0 fXpGzf\\"></div>"`;
